sparse/impl/KokkosSparse_sptrsv_solve_impl.hpp

```diff
@@ -3064,9 +3074,9 @@ void lower_tri_solve(TriSolveHandle &thandle, const RowMapType row_map,
 #endif
 
           // NOTE: we currently supports only default_layout = LayoutLeft
-          using team_policy_type = Kokkos::TeamPolicy<execution_space>;
+          using team_policy_type = Kokkos::TeamPolicy<ExecutionSpace>;
           using supernode_view_type =
-              Kokkos::View<scalar_t **, default_layout, memory_space,
+              Kokkos::View<scalar_t **, default_layout, device_t,
```

Comment:
I would argue that a memory space is more appropriate since a Kokkos::View is really a container and does not need to know where code is being executed. The only exception might be for the view initialization, but I'm not sure it would respect the device type used here?

Reply:
I had to switch to `device_t` in order to resolve link errors in builds with 2 or more host spaces enabled (e.g. `KokkosKernels_PullRequest_GNU1021`). In this call path, with two or more host spaces enabled, the linker failed to find a ETI'd `GEMV` specialization for the `_serial` test binaries since the defaulted `kokkos::device` contained a memory_space of `kokkos::threads`. Spelling out the full `kokkos::device` template params here resolved this link error. Perhaps this is a kokkos core bug?